django/contrib/auth/password_validation.py

```diff
@@ -0,0 +1,134 @@
+    for validator in settings.AUTH_PASSWORD_VALIDATORS:
+        try:
+            klass = import_string(validator['NAME'])
+            validators.append(klass(**validator.get('OPTIONS', {})))
+        except ImportError:
+            raise ImproperlyConfigured("Invalid NAME for a password validator: %s. Check "
+                                       "your AUTH_PASSWORD_VALIDATORS setting." % validator['NAME'])
+
+    return validators
+
+
+def validate_password(password, user=None):
+    """
+    Validates whether the password meets all validator requirements.
+
+    If the password is valid, returns ``None``.
+    If the password is invalid, raises ValidationError with all error messages.
+    """
+    errors = []
+    for validator in get_password_validators():
+        error = validator.validate(password, user)
+        if error:
+            errors.append(error)
+    if errors:
+        raise ValidationError(errors)
+
+
+def password_changed(password, user=None):
+    """
+    Informs all validators, that have implemented this feature, that the password has been changed.
+    """
+    for validator in get_password_validators():
+        if hasattr(validator, 'password_changed'):
+            validator.password_changed(password, user)
+
+
+def password_validators_help_texts():
+    """
+    Returns a list of all help texts of all configured validators.
+    """
+    help_texts = []
+    for validator in get_password_validators():
+        help_texts.append(validator.get_help_text())
+    return help_texts
+
+
+def password_validators_help_text_html():
+    """
+    Returns an HTML string with all help texts of all configured validators in an <ul>.
+    """
+    help_texts = password_validators_help_texts()
+    return '<ul>%s</ul>' % unordered_list(help_texts)
+
+
+class MinimumLengthValidator(object):
+    """
+    Validates whether the password has a minimum length.
+    """
+    def __init__(self, min_length=8):
+        self.min_length = min_length
+
+    def validate(self, password, user=None):
+        if len(password) < self.min_length:
+            return _("This password is too short.")
+
+    def get_help_text(self):
+        return _("Your password must contain at least %(min_length)d characters.") % {'min_length': self.min_length}
+
+
+class UserAttributeSimilarityValidator(object):
+    """
+    Validates whether the password is sufficiently different from the user's attributes.
+
+    If no specific attributes are provided, this looks at a sensible list of defaults.
+    Attributes that don't exist are gracefully ignored. Comparison is made to not only
+    the full attribute value, but also it's components, so that e.g. a password is validated
+    against either part of an e-mail address, as well as the full address.
+    """
+    def __init__(self, user_attributes=None, max_similarity=0.7):
+        self.user_attributes = user_attributes if user_attributes else ('username', 'first_name', 'last_name', 'email')
```

Comment:
If `user_attributes` is set to `()` or `[]`, the default set of attributes will be used. That may be surprising. Can you make a strict check for `user_attributes is None`?

I understand that the validator doesn't do anything then and can simply be removed from the settings in that case, but I can imagine situations where someone would control `user_attributes` (e.g. through an UI) and not `AUTH_PASSWORD_VALIDATORS`.